File: CSharp/ShopTest/Ui/CLI/ConsoleUiTextCreatorTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using Xunit;

namespace ShopTest.Ui.CLI {
    using Shop.Items;
    using Shop.Customers;
    using Shop;
    using Shop.Ui.CLI;
    public class ConsoleUiBuilderTestHelper {
        public static string LoadFromFile(string fileName) {
            FileStream fs = new FileStream(fileName, FileMode.Open);
            byte[] buffer = new byte[2048];
            int nonNulls = fs.Read(buffer, 0, buffer.Length);
            return Encoding.ASCII.GetString(buffer, 0, nonNulls);
        }
    }
    public class ConsoleUiBuilderTest {
        [Fact]
        public void ConstructMainMenu_Correct() {
            UiTextCreator uiBuilder = new ConsoleUiTextCreator();
            string actual = uiBuilder.ConstructMainMenu();
            string expected = ConsoleUiBuilderTestHelper.LoadFromFile("TestData/ConstructMainMenu");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void ConstructShopMenu_WithItems() {
            UiTextCreator builder = new ConsoleUiTextCreator();
            List<ItemBatch> items = new List<ItemBatch>();
            items.Add(new ItemBatch(new Item("Black Shirt", 15), 110));
            string result = builder.ConstructShopMenu(items);
            string expected = ConsoleUiBuilderTestHelper.LoadFromFile("TestData/ConstructShopMenu1");
            Assert.Equal(expected, result);
        }
        [Fact]
        public void ConstructCustomerMenu_WithItem() {
            UiTextCreator builder = new ConsoleUiTextCreator();

            Shop shop = new Shop();

            ItemBatch itemBatch = new ItemBatch(new Item("Black Shirt", 15), 2);
            shop.Add(itemBatch);

            Customer c = new Customer(1000);
            c.Buy(shop, "Black Shirt");
            c.Buy(shop, "Black Shirt");

            string expected = ConsoleUiBuilderTestHelper.LoadFromFile("TestData/ConstructCustomerMenu");
            string actual = builder.ConstructCustomerMenu(c);

            Assert.Equal(expected, actual);
        }
        [Fact]
        public void ConstructCustomerBuyMenu() {
            UiTextCreator builder = new ConsoleUiTextCreator();

            Shop shop = new Shop();

            Item item = new Item("Red Shoes", 15);
            ItemBatch itemBatch = new ItemBatch(item, 1);
            shop.Add(itemBatch);

            string expected = ConsoleUiBuilderTestHelper.LoadFromFile("TestData/ConstructCustomerBuyMenu");
            string actual = builder.ConstructCustomerBuyMenu(shop.Items);
            Assert.Equal(expected, actual);
        }
        [Fact]
        public void ConstructCustomerRefundMenu() {
            UiTextCreator builder = new ConsoleUiTextCreator();

            Customer c = new Customer(1000);

            Shop shop = new Shop();
            Item item = new Item("Blue Pants", 60);
            shop.Add(new ItemBatch(item, 1));

            c.Buy(shop, "Blue Pants");

            string expected = ConsoleUiBuilderTestHelper.LoadFromFile("TestData/ConstructCustomerRefundMenu");
            string actual = builder.ConstructCustomerRefundMenu(c.Items);

            Assert.Equal(expected, actual);
        }

    }
}

File: Java/TddShop/TddShopTest/src/Ui/ConsoleUiTextCreatorTest.java
package Ui;

import org.junit.Assert;
import org.junit.Test;
import static org.junit.Assert.assertEquals;
import org.junit.Assert;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import Item.*;

class ConsoleUiTextCreatorTestHelper{
    public static String LoadFromFile(String fileName) throws FileNotFoundException, IOException {
        FileInputStream fileStream = new FileInputStream(fileName);
        byte[] buffer = new byte[2048];
        int read = fileStream.read(buffer);
        fileStream.close();
        String s = new String(buffer);
        s = s.trim();
        return s;
    }
}
public class ConsoleUiTextCreatorTest {
    @Test
    public void TestTestHelper() throws IOException{
        FileOutputStream outputStream = new FileOutputStream("NewFile");
        outputStream.flush();
        outputStream.close();
        String actual = ConsoleUiTextCreatorTestHelper.LoadFromFile("TestHelperTest");
        assertEquals("Hello, There!", actual);
    }
    @Test
    public void ConstructMainMenu_Correct() throws IOException{
        ConsoleUiTextCreator textCreator = new ConsoleUiTextCreator();
        String actual = textCreator.ConstructMainMenu();
        String expected = ConsoleUiTextCreatorTestHelper.LoadFromFile("ConstructMainMenu");
        assertEquals(expected, actual);
    }
    @Test
    public void ConstructShopMenu_WithItems() throws IOException{
        ConsoleUiTextCreator textCreator = new ConsoleUiTextCreator();
        List<ItemBatch> items = new ArrayList<ItemBatch>();
        items.add(new ItemBatch(new Item("Black Shirt", 15), 110));
        String result = textCreator.ConstructShopMenu(items);
        String expected = ConsoleUiTextCreatorTestHelper.LoadFromFile("ConstructShopMenu1");
        assertEquals(expected,result);
    }
}
